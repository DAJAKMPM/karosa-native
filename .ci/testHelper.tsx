import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
