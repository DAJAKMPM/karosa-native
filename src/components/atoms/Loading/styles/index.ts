/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { theme } from "@app/styles";

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: theme.colors.transparent,
  },
});
