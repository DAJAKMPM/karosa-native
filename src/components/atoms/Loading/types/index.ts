/**
 *
 * @format
 *
 */

export type PropsType = {
  size?: number | "small" | "large";
  color?: string;
};
