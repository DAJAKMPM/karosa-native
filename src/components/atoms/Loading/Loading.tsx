/**
 *
 * Loading
 * @format
 *
 */

import React, { FC } from "react";
import { ActivityIndicator, View } from "react-native";
import type { PropsType } from "./types";
import { styles } from "./styles";
import { theme } from "@app/styles";

const Loading: FC<PropsType> = (props) => {
  const { color = theme.colors.dark20, size = "small" } = props;

  return (
    <View style={styles.container}>
      <ActivityIndicator size={size} color={color} />
    </View>
  );
};

export default Loading;
