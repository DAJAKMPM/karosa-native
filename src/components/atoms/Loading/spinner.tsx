import React from "react";
import { ActivityIndicator } from "react-native";
import { theme } from "@app/styles";

export const Spinner: React.FC = () => {
  return <ActivityIndicator size="large" color={theme.colors.primary} />;
};
