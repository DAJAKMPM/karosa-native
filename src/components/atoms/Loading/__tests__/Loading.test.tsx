import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import Loading from "../Loading";

describe("Loading Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(<Loading />);

    expect(toJSON()).toMatchSnapshot();
  });
});
