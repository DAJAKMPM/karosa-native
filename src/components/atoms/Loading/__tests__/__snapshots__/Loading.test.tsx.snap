// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#1C1C1C"
      hidesWhenStopped={true}
      size="small"
    />
  </View>
</View>
`;
